### PR TITLE
test: Verify newer Federation List schema is supported and test parsing of its data

### LINKED ISSUE #23: Use new federationlist version
### Description
We need to make sure to use the latest version of the list: https://github.com/gematik/api-vzd/blob/main/src/openapi/I_VZD_TIM_Provider_Services.yaml

### What has changed?
We need to update from version 1.3.1 to 1.3.7. The most important change is, that the domain scheme has changed. We need this new schema to make the /v1/server/findByIk endpoint work.

### PR DESCRIPTION
**Pull request by jason-famedly**
This will update the raw test data sample used, which required adjusting the domains that were tested against.

Three tests were added, to verify that data is parsed in the way expected. I found multiple servers that were not following the schema for the `timAnbieter` attribute, this is now tested for as well.

I think this:
* Fixes #23

### CODE REVIEW

File: synapse_invite_checker/invite_checker.py [hunk 1]
```diff
@@ -352,12 +361,12 @@ async def user_may_invite(
         # Step 1, check federation allow list
         inviter_id = UserID.from_string(inviter)
         inviter_domain = inviter_id.domain
-        fedlist = await self.fetch_federation_allow_list()
-        if inviter_domain not in fedlist:
+        domain_set, _ = await self.fetch_federation_allow_list()
+        if inviter_domain not in domain_set:
             self.fetch_federation_allow_list.cache_clear()
-            fedlist = await self.fetch_federation_allow_list()
+            domain_set = await self.fetch_federation_allow_list()
```
Comment (nico-famedly) [inline synapse_invite_checker/invite_checker.py:367]: this looks wrong, if you now use a tuple

File: synapse_invite_checker/invite_checker.py [hunk 2]
```diff
@@ -319,8 +327,9 @@ async def fetch_federation_allow_list(self) -> set[str]:
             msg = "Empty federation list"
             raise Exception(msg)
 
-        fedlist = FederationList.model_validate_json(jws_verify.payload)
-        return {fed.domain for fed in fedlist.domainList}
+        # Make sure to validate incoming, potentially incomplete or corrupt data
```
Comment (nico-famedly) [inline synapse_invite_checker/invite_checker.py:330]:
```suggestion
        # Validate incoming, potentially incomplete or corrupt data
```

Comment (nico-famedly) [reply synapse_invite_checker/invite_checker.py:330]: I think the other words are just filler here

File: tests/test_simple.py [hunk 3]
```diff
@@ -865,3 +873,94 @@ def test_del_contact(self) -> None:
...
+        Search for a specific domain in the federation list
+        """
+        _, raw_fed_list = await self.invchecker.fetch_federation_allow_list()
+        assert len(raw_fed_list.domainList) > 0
+
+        for domain_entry in raw_fed_list.domainList:
+            if domain_entry.domain == domain:
+                return domain_entry
+
+        return None
```
Comment (nico-famedly) [inline tests/test_simple.py:915]: You can probably just assert here instead of returning None

**Comment by nico-famedly** [comment]
Apart from the tuple confusion, this looks good enough for now, imo.

You may want to use a dataclass or some alternative to have named fields for the federation list return value, but that's something we can also tackle later.

Also it might make sense to keep the federation list tests together. You introduced a new testcase for it, but the old "is domain in list" assertion is still present in a different test. Probably makes sense to group them logically.

File: tests/test_simple.py (hunk 3 above)
Comment (jason-famedly) [reply tests/test_simple.py:915]: I'm not certain what to `assert` here, and there may be a case for a test of what to do when a domain isn't present in the future. What do you recommend?

File: synapse_invite_checker/invite_checker.py (hunk 1 above)
Comment (jason-famedly) [reply synapse_invite_checker/invite_checker.py:367]: good catch, I'll fix that up

File: synapse_invite_checker/invite_checker.py (hunk 2 above)
Comment (jason-famedly) [reply synapse_invite_checker/invite_checker.py:330]: Yes this is not a school essay. Unnecessary words are unnecessary. They can go

File: tests/test_simple.py (hunk 3 above)
Comment (nico-famedly) [reply tests/test_simple.py:915]: Well, we are always expecting to find those items in the tests which call it. So just assert for now and change it, if we ever have tests which expect None. Gets rid of some None checks at the caller side and makes the return type easier to read as well.

Comment (nico-famedly) [reply tests/test_simple.py:915]: So just `assert False, f"Not found in federation list {domain}"`

**Comment by nico-famedly** [comment]
You would move the whole `test_federation_list` into the other test class for example and replaced the local invchecker, with the self.invchecker.

File: synapse_invite_checker/invite_checker.py (hunk 1 above)
Comment (jason-famedly) [reply synapse_invite_checker/invite_checker.py]: Fixed in e64ae8f

File: synapse_invite_checker/invite_checker.py (hunk 2 above)
Comment (jason-famedly) [reply synapse_invite_checker/invite_checker.py]: fixed in e64ae8f

File: tests/test_simple.py (hunk 3 above)
Comment (jason-famedly) [reply tests/test_simple.py]: That sounds better than what I had in my head
fixed in e64ae8f

**Review by emgrav — COMMENTED** [review]
One small note: We use [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/#specification) in the other development teams. It's up to the matrix team to decide for yourselves whether you also want to do this, but at least personally I am a sucker for consistency :) It has quite a few advantages, like allowing you to easily generate good changelogs and communicating intent more clearly at a glance.

**Review by nico-famedly — APPROVED** [review]
Looks good to me, thanks!

**Review by nico-famedly — APPROVED** [review]

**Pull request merged**